### PR TITLE
add test for Conv2D gradients

### PR DESCRIPTION
**Pull request by mikowals**

### CODE REVIEW

**Comment by Shashi456** [comment]
I'm not entirely sure how this works given #397

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -110,6 +110,52 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output, expected)
     }
 
+    func testConv2DGradient() {
+        let filter =  Tensor(shape: [3, 3, 2, 4], scalars: (0..<72).map(Float.init))
+        let bias = Tensor<Float>(zeros: [4])
```
Comment (Shashi456) [inline Tests/TensorFlowTests/LayerTests.swift]: Could you add a non-trivial bias?

Comment (mikowals) [reply Tests/TensorFlowTests/LayerTests.swift]: The values of the biases don't have an impact on the gradient. The gradient is only changed by the batch size of the inputs.

**Comment by mikowals** [comment]
I see the presubmit check failed and found the error messages:

> 889 [24/52] Compiling TensorFlow LazyTensorOperation.swift
890 /swift-apis/Sources/TensorFlow/Core/LazyTensorShapeInference.swift:88:13: error: extra argument in call status)

and 

> 893 [25/52] Compiling TensorFlow LazyTensorShapeInference.swift
894 /swift-apis/Sources/TensorFlow/Core/LazyTensorShapeInference.swift:88:13: error: extra argument in call status)

This seems unrelated to this pull request and maybe the tests just need to be rerun?  Let me know if there I have misunderstood and there is something I need to do to fix it.

**Comment by rxwei** [comment]
This is because your branch is not up to date. Could you please pull from master?

**Comment by mikowals** [comment]
Pull from master done.

**Comment by rxwei** [comment]
Thanks!

**Pull request merged**